extensions/amp-install-serviceworker/0.1/amp-install-serviceworker.js

```diff
@@ -36,30 +39,66 @@ class AmpServiceWorkerInstallation extends AMP.BaseElement {
     const src = this.element.getAttribute('src');
     assertHttpsUrl(src, this.element);
 
+    /** @private {?string}  */
+    this.iframeSrc_ = null;
+
     if (isProxyOrigin(src) || isProxyOrigin(win.location.href)) {
+      const iframeSrc = this.element.getAttribute('data-iframe-src');
+      if (iframeSrc) {
+        assertHttpsUrl(iframeSrc, this.element);
+        const origin = parseUrl(iframeSrc).origin;
+        const docInfo = documentInfoFor(win);
+        const sourceUrl = parseUrl(docInfo.sourceUrl);
+        const canonicalUrl = parseUrl(docInfo.canonicalUrl);
+        user.assert(
+            origin == sourceUrl.origin ||
+            origin == canonicalUrl.origin,
+            'data-iframe-src (%s) should be a URL on the same origin as the ' +
+            'source (%s) or canonical URL (%s) of the AMP-document.',
+            origin, sourceUrl.origin, canonicalUrl.origin);
+        this.iframeSrc_ = iframeSrc;
+        this.scheduleIframeLoad_();
+      }
       return;
     }
 
-    if (originMatches(win.location.href, src)) {
+    if (parseUrl(win.location.href).origin == parseUrl(src).origin) {
```

Comment:
Will this allow a service worker to be installed on the cache via `<amp-install-serviceworker src="https://cdn.ampproject.org/something.js">`? Is there a use case for that?